src/components/DataDimension/SourceEmptyPlaceholder.js

```diff
@@ -0,0 +1,119 @@
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from 'prop-types'
+import React from 'react'
+import {
+    DIMENSION_TYPE_DATA_ELEMENT,
+    DIMENSION_TYPE_DATA_SET,
+    DIMENSION_TYPE_EVENT_DATA_ITEM,
+    DIMENSION_TYPE_PROGRAM_INDICATOR,
+    DIMENSION_TYPE_INDICATOR,
+    DIMENSION_TYPE_EXPRESSION_DIMENSION_ITEM,
+} from '../../modules/dataTypes.js'
+import styles from './styles/EmptyPlaceholder.style.js'
+
+export const SourceEmptyPlaceholder = ({
+    loading,
+    searchTerm,
+    options,
+    allItemsSelectedMessage,
+    noItemsMessage,
+    dataType,
+    dataTest,
+}) => {
+    let message = ''
+    if (allItemsSelectedMessage && !loading && options.length && !searchTerm) {
```

Comment:
Looks like `options` needs a default prop as `undefined` will break.

Reply:
`option` should always be an array (empty or not) as it's passed from the state of selector components which is initialized to empty array.
Added a `defaultProps` anyway.

Reply:
Alright